cmd/kubeadm/app/preflight/checks.go

```diff
@@ -699,15 +726,16 @@ func RunInitMasterChecks(cfg *kubeadmapi.MasterConfiguration) error {
 		DirAvailableCheck{Path: filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ManifestsSubDirName)},
 		FileContentCheck{Path: bridgenf, Content: []byte{'1'}},
 		SwapCheck{},
-		InPathCheck{executable: "ip", mandatory: true},
-		InPathCheck{executable: "iptables", mandatory: true},
-		InPathCheck{executable: "mount", mandatory: true},
-		InPathCheck{executable: "nsenter", mandatory: true},
-		InPathCheck{executable: "ebtables", mandatory: false},
-		InPathCheck{executable: "ethtool", mandatory: false},
-		InPathCheck{executable: "socat", mandatory: false},
-		InPathCheck{executable: "tc", mandatory: false},
-		InPathCheck{executable: "touch", mandatory: false},
+		InPathCheck{executable: "ip", mandatory: true, exec: execer},
+		InPathCheck{executable: "iptables", mandatory: true, exec: execer},
+		InPathCheck{executable: "mount", mandatory: true, exec: execer},
+		InPathCheck{executable: "nsenter", mandatory: true, exec: execer},
+		InPathCheck{executable: "ebtables", mandatory: false, exec: execer},
+		InPathCheck{executable: "ethtool", mandatory: false, exec: execer},
+		InPathCheck{executable: "socat", mandatory: false, exec: execer},
+		InPathCheck{executable: "tc", mandatory: false, exec: execer},
+		InPathCheck{executable: "touch", mandatory: false, exec: execer},
+		criCtlChecker,
```

Comment:
is it expected to always look for `crictl` here?

Reply:
well, if we want to encourage people to install it and do the check via the CRI, yes. And btw, it's not mandatory :)